src/hnsw/external_index.c

```diff
@@ -46,37 +49,69 @@ static char *extract_node(char               *data,
     return tape;
 }
 
-void StoreExternalIndex(Relation        index,
+void StoreExternalIndexBlockMapGroup(
+                        Relation        index,
                         usearch_index_t external_index,
                         ForkNumber      forkNum,
                         char           *data,
+                        int            *progress,
                         size_t          external_index_size,
                         int             dimension,
-                        size_t          num_added_vectors)
+                        int             first_node_index,
+                        size_t          num_added_vectors) 
 {
+    // Create empty blockmap pages for the group
+    const int number_of_blockmaps_in_group = 1 << g_blockmap_groups;
+    elog(INFO, "davkhech number_of_blockmaps_in_group %d num_added_vectors %d", number_of_blockmaps_in_group, num_added_vectors);
+    for (int blockmap_id = 0; blockmap_id < number_of_blockmaps_in_group; ++blockmap_id) {
+        Buffer buf = ReadBufferExtended(index, forkNum, P_NEW, RBM_NORMAL, NULL);
+        LockBuffer(buf, BUFFER_LOCK_EXCLUSIVE);
+
+        GenericXLogState *state = GenericXLogStart(index);
```

Comment:
This is also an issue in the current version of the `StoreExternalIndex` function.

We need to make sure committed database edits never leave the database at an undefined state.
If some failure happens in one of the iterations of this `for` loop, some blockmap pages would be successfully saved on WAL but the rest will not be, because of the failure. 
This will cause the index be permanently corrupt and unrecoverable.

We need to do `GenericXLogStart` and `GenericXLogFinish` such that when `GenericXLogFinish` is called, the database is always in a valid state. We can merge this as is and fix this separately, as other code that is currently unchanged in StoreExternalIndexBlockMapGroup is also affected by this.